project/src/main/java/com/google/autograder/servlets/helpers/API.java

```diff
@@ -27,23 +27,18 @@
 
     public static String getCurrentUserAPIAuthorization() {
         String userEmail = Database.getCurrentUserEmail();
-
         Filter userEmailFilter = new FilterPredicate("user_email", FilterOperator.EQUAL, userEmail);
-
         Query query = new Query("AccessTokenResponse").setFilter(userEmailFilter).addSort("expires_in", SortDirection.DESCENDING);
-
         PreparedQuery results = Database.query(query);
 
         AccessTokenResponse accessTokenResponse = null;
+        String authorization = null;
 
-        Entity entity = (results.asIterable().iterator().hasNext() ? results.asIterable().iterator().next() : null); ;
-
-        if (entity != null) {
+        if (results.asIterable().iterator().hasNext()) {
```

Comment:
(In general) I think you'll want to save the iterator, otherwise it may either create a new iterator or throw an exception:
```
Iterator<Entity> results = results.asIterable().iterator();
if (results.hasNext()) {
  ... results.next();
}
```

Reply:
OK, no problem.